ansicolor.js

```diff
@@ -198,131 +213,220 @@ const TEXT    = 0,
       BRACKET = 1,
       CODE    = 2
 
-function rawParse (s) {
-    
-    let state = TEXT, buffer = '', text = '', code = '', codes = []
-    let spans = []
-
-    for (let i = 0, n = s.length; i < n; i++) {
-
-        const c = s[i]
+class Span {
+  constructor(code, text) {
+    this.code = code;
+    this.text = text;
+
+    // Those are added in the actual parse, this is done for performance reasons to have the same hidden class
+    this.css = "";
+    this.color = "";
+    this.bgColor = "";
+    this.bold = undefined;
+    this.inverse = undefined;
+    this.italic = undefined;
+    this.underline = undefined;
+    this.bright = undefined;
+    this.dim = undefined;
+  }
+}
 
-        buffer += c
+// getString as function instead of string to allow garbage collection
+function* rawParse(getString) {
```

Comment:
instead of this:

```js
const str = <1 GB of string>

rawParse(str)
```

the `str` will be saved in memory until the execution finish

we get a function to get the string so when the function finish the string can be garbage collected